contrib/sddpg/core.c

```diff
@@ -169,12 +170,22 @@ static HB_ERRCODE pgsqlConnect( SQLDDCONNECTION * pConnection, PHB_ITEM pItem )
    PGconn *       pConn;
    ConnStatusType status;
    const char *   pszHost;
-
-   pszHost = hb_arrayGetCPtr( pItem, 2 );
-   if( pszHost && ( strncmp( pszHost, "postgresql://", 13 ) == 0 || strchr( pszHost, '=' ) ) )
-      pConn = PQconnectdb( pszHost );
-   else
-      pConn = PQsetdbLogin( pszHost, hb_arrayGetCPtr( pItem, 6 ), hb_arrayGetCPtr( pItem, 7 ), hb_arrayGetCPtr( pItem, 8 ), hb_arrayGetCPtr( pItem, 5 ), hb_arrayGetCPtr( pItem, 3 ), hb_arrayGetCPtr( pItem, 4 ) );
+   void ** hbConn;
+   PHB_ITEM  pSecond;
+   int existingConnection = 0;
+
+   pSecond = hb_itemArrayGet( pItem, 2);
+   if HB_IS_POINTER( pSecond )  {
+      hbConn = (void **) hb_itemGetPtr( pSecond );
+      pConn = (PGconn *) *hbConn;
+      existingConnection = 1;
+   } else {
+      pszHost = hb_arrayGetCPtr( pItem, 2 );
+      if( pszHost && ( strncmp( pszHost, "postgresql://", 13 ) == 0 || strchr( pszHost, '=' ) ) )
+          pConn = PQconnectdb( pszHost );
+      else
+          pConn = PQsetdbLogin( pszHost, hb_arrayGetCPtr( pItem, 6 ), hb_arrayGetCPtr( pItem, 7 ), hb_arrayGetCPtr( pItem, 8 ), hb_arrayGetCPtr( pItem, 5 ), hb_arrayGetCPtr( pItem, 3 ), hb_arrayGetCPtr( pItem, 4 ) );
+   }
```

Comment:
These nits need to be fixed:
- indenting to be 3 chars
- `} else {` to be split like it is done in rest of Harbour code
- double space after `pSecond )`
- missing `if(  )` enclosure in `HB_IS_POINTER()` line
- missing spaces around type in casts
- `existingConnection` should be a `HB_BOOL` and named `fExistingConncetion`
- shouldn't `PQfinish()` call be omitted if it was an existing connection?

My overall concern though is that the idea is based on a raw (non-GC-collected, non-typed) 
pointer to be received, which is unsafe and I also can't see where it can originate from, 
because there is no facility in Harbour or its contribs that'd return such raw pointer.
Probably it'd be best to use the GC-pointers inside `pgsql`, but that means a bunch of
additional development.